pkg/dao/rpm_test.go

```diff
@@ -241,14 +253,95 @@ func (s *RpmSuite) TestRpmSearch() {
 	}
 }
 
+// func (s *RpmSuite) randomPackageName(size int) string {
+func randomPackageName(size int) string {
+	const lookup string = "0123456789abcdefghijklmnopqrstuvwxyz"
+	return seeds.RandStringWithTable(size, lookup)
+}
+
+// func (s *RpmSuite) randomHexadecimal(size int) string {
+func randomHexadecimal(size int) string {
+	const lookup string = "0123456789abcdef"
+	return seeds.RandStringWithTable(size, lookup)
+}
+
+// func (s *RpmSuite) randomYumPackage() yum.Package {
+func randomYumPackage(pkg *yum.Package) {
+	if pkg == nil {
+		return
+	}
+	pkgName := randomPackageName(32)
+	pkg.Name = pkgName
+	pkg.Arch = "x86_64"
+	pkg.Summary = pkgName + " summary"
+	pkg.Version = yum.Version{
+		Version: "1.0.0",
+		Release: "dev",
+		Epoch:   0,
+	}
+	pkg.Type = "rpm"
+	pkg.Checksum = yum.Checksum{
+		Type:  "sha256",
+		Value: randomHexadecimal(64),
+	}
+}
+
+func (s *RpmSuite) preparePagedRpmInsert(scenario int, limit int) []yum.Package {
+	switch scenario {
+	case scenario0:
+		{
+			return []yum.Package{}
+		}
+	case scenario3:
+		// The reason of this scenario is to make debugging easier
+		{
+			pkgs := make([]yum.Package, 3)
+			for i := 0; i < 3; i++ {
+				randomYumPackage(&pkgs[i])
+			}
+			return pkgs
+		}
+	case scenarioUnderThreshold:
+		{
+			pkgs := make([]yum.Package, limit-1)
+			for i := 0; i < limit-1; i++ {
+				randomYumPackage(&pkgs[i])
+			}
+			return pkgs
+		}
+	case scenarioThreshold:
+		{
+			pkgs := make([]yum.Package, limit)
+			for i := 0; i < limit; i++ {
+				randomYumPackage(&pkgs[i])
+			}
+			return pkgs
+		}
+	case scenarioOverThreshold:
+		{
+			pkgs := make([]yum.Package, limit+1)
+			for i := 0; i < limit+1; i++ {
+				randomYumPackage(&pkgs[i])
+			}
+			return pkgs
+		}
+	default:
+		{
+			return []yum.Package{}
+		}
+	}
```

Comment:
It might make sense to extract the contents of each case to another method, like this:

```
func (s *RpmSuite) prepareScenarioRpms(scenario int, limit int) []yum.Package {
	switch scenario {
	case scenario0:
		{
		        return makeYumPackage(0)
		}
	case scenario3:
		// The reason of this scenario is to make debugging easier
		{
			return makeYumPackage(3)
		}
	case scenarioUnderThreshold:
		{
			return makeYumPackage(limit-1)
		}
	case scenarioThreshold:
		{
			return makeYumPackage(limit)
		}
	case scenarioOverThreshold:
		{
			return makeYumPackage(limit+1)
		}
	default:
		{
			return makeYumPackage(0)
		}
	}
```

and just define a make yumYumPackage function to do the actual making.